proton

```diff
@@ -1343,7 +1385,8 @@ class Session:
                 else:
                     f.write(" \"" + arg + "\"")
             f.write(")\n")
-            self.dump_dbg_env(f)
+            f.write(launch_client)
+            self.dump_dbg_env(f, using_launcher_service=bool(launcher_address))
             f.write("\t\"" + g_proton.wine64_bin + "\" c:\\\\windows\\\\system32\\\\steam.exe \"${@:-${DEF_CMD[@]}}\"\n")
```

Comment:
This seems to basically work, but I'm not sure whether I'm using it right.

With this `./proton` replacing the one from Proton 7.0-2, I can run

```
.../SteamLinuxRuntime_soldier/pressure-vessel/bin/steam-runtime-launch-client \
        --bus-name=com.steampowered.App312990 \
        --directory='' \
        -- \
        xterm
```

and I'll get an `xterm` in the Proton environment (`WINEPREFIX`, etc. are set). App ID 312990 is Expendabros (a promo/demo version of Broforce), which is one of my usual test games if I don't particularly care what Windows game I'm running.

I can also run commands like

```
/tmp/proton_$(id -nu)/run cmd
/tmp/proton_$(id -nu)/run winecfg
```

but they don't exit when `cmd` or `winecfg` does. It looks as though when Proton runs `wine64 c:\windows\system32\steam.exe cmd`, the `cmd` process correctly exits when closed, but the `steam.exe` process keeps running until I send `SIGTERM` to it. What's the intention of wrapping the process in `steam.exe` here?

Workaround: if you kill the `steam-runtime-launch-client` process with `SIGTERM` (for example Ctrl+Z, `bg`, `kill %%`), it will forward that signal to `steam.exe`, which will exit.

Reply:
I'm not sure that I'm understanding correctly what the purpose of `/tmp/proton_$(id -nu)/run` is, because it defaults to running the actual (Windows) game; but in the container environment, when the "main" copy of the Windows game exits (whether successfully or by crashing), we'll tear down the container, which will terminate `steam-runtime-launcher-service`, and this script will no longer be able to poke commands into it?

If we had a mode where `steam-runtime-launcher-service` can be told to keep running after the command it wraps has exited, then that would make this script more useful.

Reply:
> without the container

So is that assuming that your host library stack is "close enough" to soldier that you can run Proton/Wine binaries built on soldier using your host library stack?

If that's the case, then this one should *not* be wrapped in `steam-runtime-launch-client`, but `winedbg` probably should.

Reply:
> If we had a mode where steam-runtime-launcher-service can be told to keep running after the command it wraps has exited, which is something I plan to implement at some point, then that would make this script more useful.

I implemented this, it'll probably be in the next container beta (run with `SRT_LAUNCHER_SERVICE_STOP_ON_EXIT=0`).

Reply:
In today's revision I've stopped editing those scripts. We can't make them directly runnable from the host unless/until Proton knows how to find a path to (a recent version of) `steam-runtime-launch-client` *that is valid for the host* (the one that Proton finds in `PATH` will usually only exist at that path inside the container).

If you want these scripts to be directly runnable from the host, we'll probably have to wait for a sufficiently new `steam-runtime-launch-client` to be included in the Steam client itself, and then have the scripts invoke it as `~/.steam/root/ubuntu12_32/steam-runtime/amd64/bin/steam-runtime-launch-client`.

Alternatively, users can run them like this:

```
/path/to/steam-runtime-launch-client --list
/path/to/steam-runtime-launch-client --bus-name=... -- /tmp/proton_$(id -nu)/winedbg
```